src/Examine/LuceneEngine/Providers/LuceneSearcher.cs

```diff
@@ -109,11 +109,19 @@ public override string[] GetAllIndexedFields()
         
         protected virtual Directory GetLuceneDirectory()
         {
+
+            
             if (_nrtWriter != null)
             {
+                if (_directory != null && _nrtWriter.Directory != _directory)
```

Comment:
What is this code doing?

Reply:
@Shazwazza I added comment and because you commented I noticed I forgot to update that part to use WriterTracker.Current.MapAsOld( instead of direct dispose :)

Reply:
There is a lot of edge case logic in here to support scenarios that are totally outside of the core of Examine and I worry that these changes will bleed into creating strange underlying issues and code maintenance questions. (i.e. why is this code here if it never needs to exist to support what Examine itself is doing). My other worry is that porting this to 2.0.0 on netcore will be problematic. We don't have any WriterTracker anymore. There is a single writer for an index and it is managed by the index and disposed when it shuts down. Maybe there's a better way to support all of the changes that you are wanting to do by exposing protected properties so that you can sub class some of these objects to control things the way that you want. I would very much prefer to not have code like this in here and instead make the code better extensible to provide changing the logic outside of this library. Does that make sense?

Reply:
@Shazwazza in 2.0.0 we dont need that at all, as lucene 4.8 support duplicator, so we dont need really play around with swapping indexes :). So we can discuss what we can do that :)